### DISCUSSION TITLE
Publish Template Error

### DISCUSSION THREAD

**Discussion by aqidd** [muil-io/muil#3] [Q&A]
First of all, thank you for building this!

Here's my attempt to run the open source version:

- Clone the repo
- Execute entry.sh
- Build admin and client frontend
- Run with `yarn start`

I managed to setup the admin interface in localholst:3000 and it looks neat. The next step should be publishing the template right?

Here's what I've done:

- Following the quickstart, executing `npx -p @muil/cli muil-cli init` in a different project
- Run template viewer
- Generate API key and create .muilrc configuration with host `"host": "http://localhost:3000",`
- Run cli to publish the template

In this step, I got some errors:

- When trying to download from template viewer, this error occurs
```
Template Rendering Failed:
Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons: 1. You might have mismatching versions of React and the renderer (such as React DOM) 2. You might be breaking the Rules of Hooks 3. You might have more than one copy of React in the same app See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
```

- When trying to publish template, this error occurs
```
yarn muil-cli publish
yarn run v1.22.10
$ muil-cli publish
✔ Are you sure you want to publish current changes? … yes
 ✔ Compiling templates...
 • Uploading templates... ✘ Host http://localhost:3000/api is not accepting the request. please check host or change to valid api key
{
  statusCode: 500,
  message: 'The "path" argument must be of type string. Received undefined',
  error: 'Internal Server Error'
}
```

**Comment by shahafa** [comment]
Hi @aqidd 

Can you please share the template you are trying to publish?

**Reply by aqidd** [reply]
I'm publishing the sample files coming from template viewer.
